capnpy/__main__.py

```diff
@@ -5,54 +5,57 @@
 Options:
   --no-convert-case    Don't convert camelCase to camel_case
   --no-pyx             Always produce a .py file, even if Cython is available
+  --no-version-check   Don't check for version discrepancy.
 """
+from __future__ import print_function
 
 import sys
-import py
 import time
 import docopt
+
 from capnpy import load_schema
 from capnpy.message import load
 from capnpy.compiler.compiler import StandaloneCompiler
 
 
 def decode(args):
-    print >> sys.stderr, 'Loading schema...'
+    print('Loading schema...', file=sys.stderr)
     a = time.time()
-    mod = load_schema(args['SCHEMA'],
+    mod = load_schema(modname=args['SCHEMA'],
                       convert_case=args['--convert-case'],
                       pyx=args['--pyx'])
     b = time.time()
-    print >> sys.stderr, 'schema loaded in %.2f secs' % (b-a)
-    print >> sys.stderr, 'decoding stream...'
+    print('schema loaded in %.2f secs' % (b-a), file=sys.stderr)
+    print('decoding stream...', file=sys.stderr)
     cls = getattr(mod, args['CLASS'])
-    with open(args['FILE']) as f:
+    with open(args['FILE'], 'rb') as f:
         i = 0
         while True:
             try:
                 obj = load(f, cls)
             except ValueError:
                 break
-            print obj.shortrepr()
+            print(obj.shortrepr())
             i += 1
             if i % 10000 == 0:
-                print >> sys.stderr, i
+                print(i, file=sys.stderr, end='\r')
```

Comment:
This prints every 10000 messages. With end='\r' it overwrites the previous printed number, so it reduces clutter. Its better to have one line saying 1000000 than a 100 lines counting up, imo